src/server/job_recov_db.c

```diff
@@ -519,60 +339,138 @@ job_recov_db_spl(pbs_db_job_info_t *dbjob)
  *	Recover job from database
  *
  * @param[in]	jid - Job id of job to recover
+ * @param[in]	pjob - job pointer, if any, to be updated
  *
  * @return      The recovered job
  * @retval	 NULL - Failure
  * @retval	!NULL - Success, pointer to job structure recovered
  *
  */
 job *
-job_recov_db(char *jid)
+job_recov_db(char *jid, job *pjob)
 {
-	job		*pj = NULL;
 	pbs_db_job_info_t dbjob = {{0}};
 	pbs_db_obj_info_t obj;
+	int rc = -1;
 	pbs_db_conn_t *conn = svr_db_conn;
 
-	obj.pbs_db_obj_type = PBS_DB_JOB;
-	obj.pbs_db_un.pbs_db_job = &dbjob;
+	if (pjob)
+		strcpy(dbjob.ji_savetm, pjob->ji_savetm);
+	else
+		dbjob.ji_savetm[0] = '\0';
+	
+	strcpy(dbjob.ji_jobid, jid);
 
-	if (pbs_db_begin_trx(conn, 0, 0) !=0)
-		goto db_err;
+	rc = pbs_db_load_obj(conn, &obj);
+	if (rc == -2)
+		return pjob; /* no change in job, return the same job */
 
-	strcpy(dbjob.ji_jobid, jid);
+	if (rc == 0) {
+		pjob = job_recov_db_spl(&dbjob, pjob);
```

Comment:
Please remove the branches as it is single line statement.

Reply:
done